.circleci/config.yml

```diff
@@ -502,6 +502,32 @@ commands:
           port: 5556
           timeout: 60
 
+  compile-all-jars:
+    description: Compile and run checkstyle on pepper, housekeeping, and study-builder
+    parameters:
+      m2_repo:
+        type: string
+        default: *m2_repo_path
+      repo_path:
+        type: string
+        default: *repo_path
+      restore_m2_cache:
+        type: boolean
+        default: false
+    steps:
+      - when:
+          condition: << parameters.restore_m2_cache >>
+          steps:
+            - restore_cache: *m2_cache_keys
+      - run:
+          name: Compiling pepper-apis, housekeeping, and study-builder
+          command: |
+            mvn install --batch-mode -DskipTests -Dmaven.repo.local=<< parameters.m2_repo >>
+            mvn install -f parent-pom.xml --batch-mode -DskipTests -Dmaven.repo.local=<< parameters.m2_repo >>
+            mvn package -f housekeeping-pom.xml --batch-mode -DskipTests -Dmaven.repo.local=<< parameters.m2_repo >>
+            cd << parameters.repo_path >>/study-builder
+            mvn package --batch-mode -DskipTests -Dmaven.repo.local=<< parameters.m2_repo >>
+
```

Comment:
Does running `install` catch problems? If not we should just run `test-compile` as creating the mega jars is slow (was trying to keep the many-times-a-day/be-done-already run tests as fast as possible). Had been deferring building the jars on purpose.
Also was shooting for keeping the number `mvn` invocations to a minimum. take time.
And checkstyle execution was only happening on the non-parallel compile-run-test-suite-job . That job has less to do/finishes faster and checkstyle was touching everything there.

Reply:
`install` doesn't catch additional problems, but I think we have to do it given the fact that study-builder is a separate thing and needs the pepper-apis dependency in .m2. I can probably turn the other `package` ones into `test-compile`.

About the parallel job: initially I didn't want to put this in that job at all. We don't need two jobs doing the same compile task. But I was thinking if there was compile/checkstyle error then the parallel job will catch it too late in the game. Maybe that's a non-concern? Should I just remove it from that job?

Reply:
I think we can take it out of the parallel-*-job. Does not add value to have it there too and is the slower job. Doing compile-all-jars only in the compile-and-run-test-suite-jobs likely to keep the overall build time the same. Hopefully the caching will be OK.

Reply:
On second thought, I think `compile-all-jars` does add value in the parallel job. We need to do a compile anyways for the parallel job to run the tests, but now we're moving it up earlier in the process. This way, if there was any compile/checkstyle issues, we don't even need to do the parallel test thingy, so we fail faster.

Reply:
Don't think this is deal breaker, but probably does not make it faster. Since the `parallel-compile-and-test-job`  job runs in parallel to the `compile-and-run-test-suite-job` that would include the `compile-all-jars step` ,  both jobs would find the problem around the same time, but then `parallel-compile-and-test-job` would take longer to complete.
We probably splitting hairs anyways. Probably just an addition 20 or 30 secs. But on the other hand sometimes we can get impatient.